imageregistry/v1/types.go

```diff
@@ -303,8 +302,32 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access level for the storage account.
+	// "Internal" means the storage account will be private, "External" means
+	// the storage account will be publicly accessible.
+	// Defaults to "External".
+	// +kubebuilder:validation:Enum="Internal";"External"
+	// +kubebuilder:default="External"
+	// +optional
+	NetworkAccess AzureNetworkAccessibility `json:"networkAccess,omitempty"`
```

Comment:
So far the storage account has been publicly accessible. Do `AccountName`, `Container` and `CloudName` are part of the public access configuration? When the private access is configured, do the existing fields play any role in the configuration? I.e. Are the existing fields required to be set alongside the new fields (`VNetName`, `SubnetName`, `PrivateEndpointName`)?

Reply:
The image registry status reflects the entire ImageRegistryConfigStorage struct. Is the expectation here that whatever gets configured in the config will get 1:1 reflect in the status?

Reply:
`AccountName`, `Container` and `CloudName` are part of both public and private configurations. `AccountName` is used to link the private endpoint with the storage account. BUT  if users don't provide them, the operator will generate these fields (`AccountName`, `Container`) when it creates the assets in Azure, then write them in the spec. `CloudName` I believe comes from the `Infrastructure` object.

Reply:
> Is the expectation here that whatever gets configured in the config will get 1:1 reflect in the status?

Yes, that is how the currently operator works.

Reply:
Given all three (`VNetName`, `SubnetName`, `PrivateEndpointName`) fields configures the endpoint they should be located under the same new data type. E.g. RegistryEndpointNetworkConfig (naming up to you). With validation rules saying whether all three fields need to be specified, or which field is always required. Or, whether there's some defaulting. E.g. use default vnet/subnet.

Also, is network access always going to have only two modes? Public/Private? Or, is there a chance another option will get added later? [Discriminated unions|https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions] are useful for defining mutually exclusive blocks of configs.
Public network access currently does not require additional fields. Only `NetworkAccess` set. Is there a chance you could introduce a field for public access that's not shared with the private access configuration?

Reply:
Should the new fields be feature gated as a TP? https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#tech-preview for more info.

Reply:
> Also, is network access always going to have only two modes? Public/Private?

I don't foresee a case where we would expand network access to anything other than public/private. 
Public network access is the default in Azure and the Azure api does not need any extra information from us for that to work... I don't think we need a discriminated union, tho I guess it wouldn't hurt. I think the new types should cover it - PTAL